client/components/domains/connect-domain-step/transfer-domain-step-auth-code.jsx

```diff
@@ -1,8 +1,38 @@
-import React from 'react';
-import ConnectDomainStepOwnershipAuthCode from 'calypso/components/domains/connect-domain-step/connect-domain-step-ownership-auth-code';
+import { withShoppingCart } from '@automattic/shopping-cart';
+import { useI18n } from '@wordpress/react-i18n';
+import React, { useCallback } from 'react';
+import { connect } from 'react-redux';
+import { transferDomainAction } from 'calypso/components/domains/use-my-domain/utilities';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import DomainStepAuthCode from './domain-step-auth-code';
 
-const TransferDomainStepAuthCode = ( props ) => {
-	return <ConnectDomainStepOwnershipAuthCode { ...props } />;
+import './style.scss';
+
+const TransferDomainStepAuthCode = ( {
+	className,
+	domain,
+	pageSlug,
+	transferDomainActionHandler,
+	progressStepList,
+} ) => {
+	const { __ } = useI18n();
+	const recordTransferButtonClickInUseYourDomain = useCallback(
+		() => recordTracksEvent( 'calypso_use_your_domain_transfer_click', { domain } ),
+		[ domain ]
+	);
+	return (
+		<DomainStepAuthCode
+			buttonMessage={ __( 'Check readiness for transfer' ) }
+			className={ className }
+			domain={ domain }
+			onBeforeValidate={ recordTransferButtonClickInUseYourDomain }
+			validateHandler={ transferDomainActionHandler }
+			pageSlug={ pageSlug }
+			progressStepList={ progressStepList }
```

Comment:
I guess we'll need to set the proper `authCodeDescription` prop here.

Reply:
Since this prop has the same value on both specialized components, I opted to use the component's `defaultProps` value for that.

Reply:
The first sentence is not supposed to be part of the transfer flow.

Reply:
Oops. OK... That one passed. Let me fix it.